boards/qemu_rv32_virt/src/main.rs

```diff
@@ -0,0 +1,431 @@
+//! Board file for qemu-system-riscv32 "virt" machine type
+
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+
+use capsules::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::component::Component;
+use kernel::hil;
+use kernel::hil::time::Alarm;
+use kernel::Chip;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+use qemu_rv32_virt_chip::chip::{QemuRv32VirtChip, QemuRv32VirtDefaultPeripherals};
+use rv32i::csr;
+
+pub mod io;
+
+pub const NUM_PROCS: usize = 4;
+
+// Actual memory for holding the active process structures. Need an empty list
+// at least.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] =
+    [None; NUM_PROCS];
+
+// Reference to the chip for panic dumps.
+static mut CHIP: Option<
+    &'static QemuRv32VirtChip<
+        VirtualMuxAlarm<'static, sifive::clint::Clint<'static>>,
+        QemuRv32VirtDefaultPeripherals,
+    >,
+> = None;
+
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+/// Dummy buffer that causes the linker to reserve enough space for the stack.
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x2000] = [0; 0x2000];
+
+// Dummy client, printing information about received packets
+// struct VirtIONetDummyClient(&'static capsules::virtio::devices::virtio_net::VirtIONet<'static>);
+// impl capsules::virtio::devices::virtio_net::VirtIONetClient for VirtIONetDummyClient {
+//     fn packet_received(&self, id: usize, buffer: &'static mut [u8], len: usize) {
+//         debug!("Packet received from virtio-net if {}, len {}.", id, len);
+//         self.0.return_rx_buffer(buffer);
+//     }
+
+//     fn packet_sent(&self, _id: usize, _buffer: &'static mut [u8]) {
+//         unimplemented!("VirtIONetDummyClient packet_sent");
+//     }
+// }
```

Comment:
Can this just be removed? Commented out code is prone to bit rotting.

Reply:
I don't expect it to stay in here long. It's useful if you just want to test the network interface's basic functionality until we have a proper HIL and driver/stack.